predicators/spot_utils/spot_utils.py

```diff
@@ -426,6 +474,90 @@ def get_apriltag_pose_from_camera(
+        source_depth: str,
+    ) -> Dict[str, Tuple[float, float, float]]:
+        """Get object location in 3D (no orientation) estimated using
+        pretrained SAM model.
+
+        Args:
+            class_name: name of object class
+        """
+        # Only support using depth image to obatin location
+        image_request = [
+            build_image_request(source,
+                                pixel_format=None if
+                                ('hand' in source or 'depth' in source) else
+                                image_pb2.Image.PIXEL_FORMAT_RGB_U8)
+            for source in [source_rgb, source_depth]
+        ]
+        image_responses = self.image_client.get_image(image_request)
+
+        image = {
+            'rgb': process_image_response(image_responses[0]),
+            'depth': process_image_response(image_responses[1]),
+        }
+        image_responses = {
+            'rgb': image_responses[0],
+            'depth': image_responses[1],
+        }
+
+        res_locations = get_object_locations_with_sam(
+            classes=[class_name],
+            res_image=image,
+            res_image_responses=image_responses,
+            source_name=source_rgb,
+            plot=CFG.spot_visualize_vision_model_outputs)
+
+        # We only want the most likely sample (for now).
+        # NOTE: we make the hard assumption here that
+        # we will only see one instance of a particular object
+        # type. We can relax this later.
+        if len(res_locations) > 0:
+            assert len(res_locations) == 1
+            camera_tform_body = get_a_tform_b(
+                image_responses['depth'].shot.transforms_snapshot,
+                image_responses['depth'].shot.frame_name_image_sensor,
+                BODY_FRAME_NAME)
+            object_rt_gn_origin = self.convert_obj_location(
+                camera_tform_body, *res_locations[0])
+
+            # Use the input class name as the identifier for object(s) and
+            # their positions
+            return {class_name: object_rt_gn_origin}
+
+        return {}
+
+    def convert_obj_location(
+            self, camera_tform_body: bosdyn.client.math_helpers.SE3Pose,
+            x: float, y: float, z: float) -> Tuple[float, float, float]:
+        """Given an x, y, z position in the camera frame, transform it into the
+        map frame."""
+        body_tform_object = (camera_tform_body.inverse()).transform_point(
+            x, y, z)
+        # Get graph_nav to body frame.
+        state = self.get_localized_state()
+        gn_origin_tform_body = math_helpers.SE3Pose.from_obj(
+            state.localization.seed_tform_body)
+        # Apply transform to object to body location
+        object_rt_gn_origin = gn_origin_tform_body.transform_point(
+            body_tform_object[0], body_tform_object[1], body_tform_object[2])
+        return object_rt_gn_origin
+
+    @staticmethod
+    def rotate_image(image: Image, source_name: str) -> Image:
```

Comment:
should this be using the hardcoded rotation values from the other file?

Reply:
turns out this function is pointless lol; i removed